src/clj_holmes/rules/loader.clj

```diff
@@ -54,25 +62,62 @@
   (and (.isFile file)
        (-> file .getName (string/ends-with? ".yml"))))
 
-(defn ^:private read-rules [^String directory]
-  (let [reader (comp first yaml/parse-string slurp)]
-    (->> directory
-         File.
-         file-seq
-         (filter is-rule?)
-         (map reader)
-         set)))
+(defn ^:private OrderedMap->Map [rule]
+  (walk/postwalk (fn [object]
+                   (if (instance? OrderedMap object)
+                     (into {} object)
+                     object))
+                 rule))
+
+(defn ^:private rule-reader [^File rule-path]
+  (let [read-rule (->> rule-path
+                       slurp
+                       yaml/parse-string
+                       first
+                       OrderedMap->Map)]
+    (with-meta read-rule {:rule-path (-> rule-path .getAbsoluteFile str)})))
 
 (defn ^:private filter-rule-by-tags [rule-tags rules]
   (if (seq rule-tags)
     (filter (fn [rule]
               (let [existing-rule-tags (get-in rule [:properties :tags])]
-                (boolean (some (set existing-rule-tags) rule-tags))))
+                (-> existing-rule-tags
+                    set
+                    (some rule-tags)
+                    boolean)))
             rules)
     rules))
 
+(defn ^:private check-if-rule-is-valid [rule]
```

Comment:
This function could probably be more aptly named something like `run-spec-validation`?